btree/btree.c

```diff
@@ -3554,8 +3552,13 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 				if (bop->bo_flags & BOF_LOCKALL) {
 					lock_op_unlock(bop->bo_arbor->t_desc);
 					return fail(bop, -ETOOMANYREFS);
-				} else
+				} else {
```

Comment:
you need to reset the oi->i_trial to 0 here when setting the BOF_LOCKALL flag.

Reply:
This **if (bop->bo_flags & BOF_LOCKALL) {** is incorrect. Instead you should have an assert to make sure that if path_check fails then BOF_LOCKALL is NOT set. If the path changes even with LOCKALL then we have some serious issue which we need to know of during our debugging.

Reply:
I will add assert if we `path_check` fails when `BOF_LOCKALL` flag is set. 
I think it will be better to not reset `oi->i_trial` to 0 when setting the BOF_LOCKALL flag. In case of any failure we can refer to this value if needed.